tests/test_templates.py

```diff
@@ -0,0 +1,752 @@
+
+            # Initial state preparation
+            for wire in wires:
+                qml.Hadamard(wires=wire)
+
+            # Grover's iterator
+            for _ in range(iterations):
+                qml.FlipSign(omega, wires=wires)
+                qml.templates.GroverOperator(wires)
+
+            return qml.probs(wires=wires)
+
+        prob = circuit(omega)
+        index = omega.astype(int)
+        index = functools.reduce(
+            lambda sum, x: sum + (1 << x[0]) * x[1],
+            zip([i for i in range(len(index) - 1, -1, -1)], index),
+            0,
+        )
+        assert np.allclose(np.sum(prob), 1.0)
+        assert prob[index] > 0.95
+        assert np.sum(prob) - prob[index] < 0.05
+
+
+class TestAngleEmbedding:
+    """Test the AngleEmbedding algorithm."""
+
+    @pytest.mark.parametrize("n_qubits", range(2, 12, 2))
+    def test_angleembedding(self, n_qubits):
+        dev = qml.device(device_name, wires=n_qubits)
+        dq = qml.device("default.qubit")
+
+        def circuit(feature_vector):
+            qml.AngleEmbedding(features=feature_vector, wires=range(n_qubits), rotation="Z")
+            qml.Hadamard(0)
+            return qml.state()
+
+        X = np.random.rand(n_qubits)
+
+        res = qml.QNode(circuit, dev, diff_method=None)(X)
+        ref = qml.QNode(circuit, dq, diff_method=None)(X)
+
+        assert np.allclose(res, ref)
+
+
+class TestAmplitudeEmbedding:
+    """Test the AmplitudeEmbedding algorithm."""
+
+    @pytest.mark.parametrize("n_qubits", range(2, 12, 2))
+    def test_amplitudeembedding(self, n_qubits):
+        dev = qml.device(device_name, wires=n_qubits)
+        dq = qml.device("default.qubit")
+
+        def circuit(f=None):
+            qml.AmplitudeEmbedding(features=f, wires=range(n_qubits))
+            return qml.state()
+
+        X = np.random.rand(2**n_qubits)
+        X /= np.linalg.norm(X)
+        res = qml.QNode(circuit, dev, diff_method=None)(f=X)
+        ref = qml.QNode(circuit, dq, diff_method=None)(f=X)
+
+        assert np.allclose(res, ref)
+
+
+class TestBasisEmbedding:
+    """Test the BasisEmbedding algorithm."""
+
+    @pytest.mark.parametrize("n_qubits", range(2, 12, 2))
+    def test_basisembedding(self, n_qubits):
+        dev = qml.device(device_name, wires=n_qubits)
+        dq = qml.device("default.qubit")
+
+        def circuit(feature_vector):
+            qml.BasisEmbedding(features=feature_vector, wires=range(n_qubits))
+            return qml.state()
+
+        X = np.ones(n_qubits)
+
+        res = qml.QNode(circuit, dev, diff_method=None)(X)
+        ref = qml.QNode(circuit, dq, diff_method=None)(X)
+
+        assert np.allclose(res, ref)
+
+
+class TestDisplacementEmbedding:
+    """Test the DisplacementEmbedding algorithm."""
+
+    @pytest.mark.parametrize("n_qubits", range(2, 12, 2))
+    @pytest.mark.parametrize("template", [qml.DisplacementEmbedding, qml.SqueezingEmbedding])
```

Comment:
This is testing also `SqueezingEmbedding`, right? Maybe we could rename the class and docstring.